src/plugins/FetchStickerDataPlugin.ts

```diff
@@ -0,0 +1,130 @@
+import stickers from '../../stickers.json';
+import axios from 'axios';
+import protobuf from 'protobufjs';
+import * as R from 'ramda';
+import fs from 'fs';
+import crypto from 'crypto';
+import hkdf from 'js-crypto-hkdf';
+
+import {
+  StickerPackJson,
+  StickerPackDataJson,
+  StickerPackManifest
+} from 'etc/types';
+import ErrorWithCode from 'lib/error';
+
+// ----- Locals ----------------------------------------------------------------
+
+/**
+ * Module-local gRPC client used to parse sticker pack manifests from the Signal
+ * CDN.
+ */
+const StickersProto = fs.readFileSync('src/etc/Stickers.proto');
+const protobufClient = protobuf.parse(StickersProto).root;
+
+/**
+ * See: https://developer.mozilla.org/en-US/docs/Web/API/SubtleCrypto/deriveKey
+ */
+async function deriveKeys(encodedKey: string) {
+  const masterKey = Buffer.from(encodedKey, 'hex');
+  const hash = 'SHA-256';
+  const length = 512;
+  const info = 'Sticker Pack';
+  const salt = new ArrayBuffer(32);
+  const derivedKey = (await hkdf.compute(masterKey, hash, length, info, salt)).key;
+  return [derivedKey.slice(0, 32), derivedKey.slice(32, 64)];
+}
+
+/**
+ * Decrypts a manifest returned from the Signal API using a sticker pack's
+ * pack key, provided from stickers.json.
+ */
+async function decryptManifest(encodedKey: string, rawManifest: any) {
```

Comment:
Unfortunately, we can't use the logic in `crypto.ts` to do this work as it's based on `window.crypto` which isn't available server side during the build process.